[FEATURE] - Update to kubernetes 1.20, add containerd/docker separation (#14)
sighupio/fury-kubernetes-on-premises: katalog/vsphere-cm/README.md

# vSphere controller manager

> This katalog deploy the vSphere controller manager on a VMware Kubernetes cluster backed by VCenter.

Reviewer: ```suggestion
This katalog deploys the [vSphere controller manager](https://github.com/kubernetes/cloud-provider-vsphere) on a VMware Kubernetes cluster backed by vCenter.
```

## Requirements

- Kubernetes = `1.20.x`
- Kustomize >= `v3.5.3`
- control plane and the nodes must be provisioned with cloud-provider `external` on `kubeadm.yaml`
- `disk.EnableUUID=1` on all nodes.
- VM Hardware should be at version 15 or higher.
- VMware tools installed on all nodes.

> - `/etc/hostname` must match the name of the nodes, so in our case the fqdn.

Reviewer: ```suggestion
- `/etc/hostname` must match the name of the nodes, so in our case the FQDN.
```

> - VCenter >= `v67u3`.

Reviewer: ```suggestion
- vCenter >= `v67u3`.
```

## Image repository and tag

> * vsphere cloud controller manager image: `gcr.io/cloud-provider-vsphere/cpi/release/manager:v1.20.0`

Reviewer: ```suggestion
* vSphere cloud controller manager image: `gcr.io/cloud-provider-vsphere/cpi/release/manager:v1.20.0`
```

apiVersion: v1
kind: ConfigMap
metadata:
  name: vsphere-cloud-config
  namespace: kube-system
data:
  vsphere.conf: |-
    # Global properties in this section will be used for all specified vCenters unless overriden in VirtualCenter section.
    global:
      port: 443
      insecureFlag: true

    # vcenter section
    vcenter:
      10.20.30.40:
        server: 10.20.30.40
        user: k8s@vsphere.local
        password: "REDACTED"
        datacenters:
          - Datacenter
```

> The user to be able to interact with VCenter and to be able to provision volumes dynamically needs these permissions:

Reviewer: ```suggestion
The user to be able to interact with vCenter and to be able to provision volumes dynamically needs these permissions:
```

ProviderID: vsphere://42049175-beac-93eb-b6cb-5a827184f1e3
```

> Now you cluster is ready run workloads.

Reviewer: ```suggestion
Now your cluster is ready to run workloads.
```
